extensions/ql-vscode/src/definitions.ts

```diff
@@ -0,0 +1,202 @@
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import * as tmp from 'tmp';
+import * as vscode from "vscode";
+import { decodeSourceArchiveUri, zipArchiveScheme } from "./archive-filesystem-provider";
+import { EntityValue, getResultSetSchema, LineColumnLocation, UrlValue, ColumnKindCode } from "./bqrs-cli-types";
+import { CodeQLCliServer } from "./cli";
+import { DatabaseItem, DatabaseManager } from "./databases";
+import * as helpers from './helpers';
+import { CachedOperation } from './helpers';
+import * as messages from "./messages";
+import { QueryServerClient } from "./queryserver-client";
+import { compileAndRunQueryAgainstDatabase, QueryWithResults } from "./run-queries";
+
+/**
+ * Run templated CodeQL queries to find definitions and references in
+ * source-language files. We may eventually want to find a way to
+ * generalize this to other custom queries, e.g. showing dataflow to
+ * or from a selected identifier.
+ */
+
+const TEMPLATE_NAME = "selectedSourceFile";
+const SELECT_QUERY_NAME = "#select";
+
+enum KeyType {
+  DefinitionQuery, ReferenceQuery
```

Comment:
For easier debugging, I like to do:

```
DefinitionQuery = 'DefinitionQuery',
ReferenceQuery = 'ReferenceQuery'
```

Reply:
Sure, can do. Iirc with ts enums, you do have maps in both directions (e.g. `KeyType.ReferenceQuery = 1` and `KeyType[1] = "ReferenceQuery"`) but I can understand not wanting to have the extra step.